bin/generate-gutenberg-php.php

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env php
+<?php
+/**
+ * Generates the production (plugin build) version of `gutenberg.php`,
+ * containing alternate `define` statements from the development version.
+ *
+ * @package gutenberg-build
+ */
+
+$f = fopen( dirname( dirname( __FILE__ ) ) . '/gutenberg.php', 'r' );
+
+$plugin_version = null;
+$inside_defines = false;
+
+/**
+ * Prints `define` statements for the production version of `gutenberg.php`
+ * (the plugin entry point).
+ */
+function print_production_defines() {
+	global $plugin_version;
+
+	echo "define( 'GUTENBERG_VERSION', '$plugin_version' );\n";
```

Comment:
> Is it just because we need to replace the development mode constant so we might as well?

There are a couple of other reasons as well:

- This is a simpler approach to #440.  See #585 for another way to do it which I would not be as comfortable merging.
- Tying each plugin release to a particular git commit is a good idea to make sure that we can reliably identify where each build came from.  This should also be done using git tags, for example, but a bit of redundancy here is a good thing.